evalml/problem_types/problem_types.py

```diff
@@ -21,6 +21,10 @@ def __str__(self):
                              ProblemTypes.TIME_SERIES_REGRESSION.name: "time series regression"}
         return problem_type_dict[self.name]
 
+    @classproperty
```

Comment:
Need this so that users can specify the problem type as "time series regression" (which matches the enum value) as opposed to "time_series_regression".

Reply:
Ah got it. Where were the underscores coming from previously?

Reply:
Oh is this because `ProblemTypes.TIME_SERIES_REGRESSION.name` is `"time_series_regression"`, whereas `ProblemTypes.TIME_SERIES_REGRESSION.value` is `"time series regression"`?

Reply:
Yes exactly! By default, you can't look up the enum by `.value`, only the `.name`, but we prefer users to not have to use underscores to keep it consistent with the `.value`!